recipes/b2/portable/conanfile.py

```diff
@@ -53,49 +56,89 @@ def validate(self):
             raise ConanInvalidConfiguration(
                 "Option toolset 'cxx' and 'cross-cxx' requires 'use_cxx_env=True'")
 
+    def layout(self):
+        conan.tools.layout.basic_layout(self, src_folder="source")
+
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  strip_root=True, destination="source")
+        conan.tools.files.get(
+            self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    @property
+    def b2_dir(self):
```

Comment:
```suggestion
    def _b2_dir(self):
```
Please, use protected scope for any custom method. It avoids any possible name collision in future Conan versions.